Update regex, adding simple local cache

Monter opened the pull request.

Monter committed:
Update regex and deactivate proxy in admin area

Monter committed:
Adding simple local cache

Monter committed:
Revert plugin working in admin area

Rudloff commented:
Hello,

Thanks for your patch!

Before I merge it, could you please [squash the commits with git rebase](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)?
Could you also use [PSR-2](http://www.php-fig.org/psr/psr-2/) indenting (4 spaces)?

Rudloff commented on gravatar-proxy.php:
```diff
@@ -29,18 +29,15 @@
-        $matches[0], plugin_dir_url(__FILE__).
-        'proxy.php?query='.urlencode($matches[2]), $avatar
+        $matches['0'], plugin_dir_url(__FILE__).
+        'proxy.php?query='.urlencode($matches['2']), $avatar
```
I'm not sure what is point of this change.

Monter replied:
No, because the proxy.php file not refer to WordPress functions. Functions available in WordPress will work only in the main file (gravatar-proxy.php)

Monter replied:
It is a remnant of the other ideas for a modification, for a while it'll correct :-)

Rudloff commented on proxy.php:
```diff
@@ -11,9 +11,25 @@
  * @license  GPL http://www.gnu.org/licenses/gpl.html
  * @link     https://rudloff.pro/
  */
+$path =  realpath(dirname(__FILE__));
```
Wouldn't it be cleaner to use `plugin_dir_path(__FILE__)`?

Monter replied:
No, because the proxy.php file not refer to WordPress functions. Functions available in WordPress will work only in the main pliugin file (gravatar-proxy.php)

Monter commented:
Hmm, It will be a problem, because I do not use git. I operate exclusively through this website.

I'm not a professional programmer, I always use indentation about the size of two spaces. Is that a problem?

Monter committed:
Update gravatar-proxy.php

Monter committed:
PSR-2 indenting

Monter committed:
Don't run on the admin options-discussion page

Simpler would be to use the get_current_screen (), but it works only on the sides of the administrative panel...

Monter committed:
Adding cache cleaning option

The files in the cache are cleaned (and downloaded again) if they are older than 30 days (2592000 seconds)

Rudloff commented:
Thanks!
I have merged your commits in develop.

I might replace the cache part with something like [PHP SimpleCache](https://github.com/gilbitron/PHP-SimpleCache) though.

Monter commented:
Instead of a few simple lines of code to create a working cache You have added a lot of files doing exactly the same thing, though in total I do not see the option of cleaning the cache.
Your code, your choice. For me it makes no sense...
Regards.

Rudloff commented:
[Code reuse](https://en.wikipedia.org/wiki/Code_reuse) is a good pratice. Instead of reinventing the wheel, we can rely on shared, proven code provided by libraries.

The library is installed with [Composer](https://getcomposer.org/) so normally I wouldn't even have to include it in the repository. But unfortunately, [WordPress does not play well with Composer](http://wptavern.com/a-narrative-of-using-composer-in-a-wordpress-plugin) so we have to include the full vendor directory in the repository.

(Also, PHP SimpleCache does clean the cache every 3600 seconds by default.)

Monter commented:
Good practice is: "less is more". Especially in the case of oversized and ponderous WordPress :-)
So, I prefer "my" version.
PS. nice that we finally added the cache does not mention that this idea did not you fell.
